numpy/linalg/linalg.py

```diff
@@ -2021,14 +2021,8 @@ def lstsq(a, b, rcond="warn"):
         work = zeros((lwork,), t)
         results = lapack_routine(m, n, n_rhs, a, m, bstar, ldb, s, rcond,
                                  0, work, -1, rwork, iwork, 0)
-        lwork = int(abs(work[0]))
-        rwork = zeros((lwork,), real_t)
-        a_real = zeros((m, n), real_t)
-        bstar_real = zeros((ldb, n_rhs,), real_t)
-        results = lapack_lite.dgelsd(m, n, n_rhs, a_real, m,
```

Comment:
I must admit that I don't understand why this call for size information on the regular, non-complex routine was made. I suspect it was to work around a bug in `zgelsd`, which may well have been fixed with the update to lapack-lite.

In any case, with my changes, the routine just uses the results of the first call, which should be correct.